### PR TITLE
ACT-1091: Add entity type dictionary/elements support

### PR DESCRIPTION
**Pull request by jpartin2**
Adding support for entity type dictionaries and entity type dictionary terms.

### CODE REVIEW

**Review by gotvino — APPROVED** [review]

File: src/sdk/models/entityTypes.ts [hunk 1]
```diff
@@ -7,10 +7,21 @@ export interface EntityType extends IEntityIdentifier<string>, IAuditedEntity {
...
 	threshold?: number;
+	keywords?: EntityTypeDictionary;
 	readonly?: boolean;
 	disabled?: boolean;
 }
 
+export interface EntityTypeDictionary extends IEntityIdentifier<string> {
+	name?: string;
+	description?: string;
+	readonly?: boolean;
```
Comment (smbecker) [inline src/sdk/models/entityTypes.ts:18]: `elements?: EntityTypeDictionaryElement[]`

**Review by smbecker — CHANGES_REQUESTED** [review]
Minor change for when we add dictionaries.

File: src/sdk/models/entityTypes.ts [hunk 2]
```diff
@@ -7,10 +7,22 @@ export interface EntityType extends IEntityIdentifier<string>, IAuditedEntity {
 	kind?: 'pattern' | 'block_list' | 'transform' | 'classifier' | string;
 	category?: Category;
 	threshold?: number;
+	keywords?: EntityTypeDictionary;
```
Comment (avolinsky) [inline src/sdk/models/entityTypes.ts:10]: Same question as https://github.com/portalarchitects/skysync-cli/pull/181 Are we not going to create multiple EntityType classes like on back end? `keywords` property does not apply to `TransformEntityType`

cc: @bentatum, @gotvino, @smbecker

**Comment by jpartin2** [comment]
@gotvino @smbecker @avolinsky @bentatum  Added the distinct types for block list and pattern entity types. I added in Ben's regex patterns PR changes to this one just to build out the concept.

Please have a look and let me know any thoughts.

File: src/sdk/models/entityTypes.ts [hunk 3]
```diff
@@ -11,6 +11,44 @@ export interface EntityType extends IEntityIdentifier<string>, IAuditedEntity {
 	disabled?: boolean;
 }
 
+export interface BlockListEntityType extends EntityType {
+	kind: 'block_list';
```
Comment (jpartin2) [inline src/sdk/models/entityTypes.ts:15]: Did we want the discriminators on these? Seems to work well but the jobs examples (TransferJob, PolicyJob, etc.) don't seem to do this.

**Review by avolinsky — APPROVED** [review]

**Pull request merged**